src/services/refactors/extractType.ts

```diff
@@ -205,21 +233,28 @@ function flattenTypeLiteralNodeReference(checker: TypeChecker, node: TypeNode |
         }
         return result;
     }
-    else if (isParenthesizedTypeNode(node)) {
-        return flattenTypeLiteralNodeReference(checker, node.type);
+    else if (isParenthesizedTypeNode(selection)) {
+        return flattenTypeLiteralNodeReference(checker, selection.type);
     }
-    else if (isTypeLiteralNode(node)) {
-        return node.members;
+    else if (isTypeLiteralNode(selection)) {
+        return selection.members;
     }
     return undefined;
 }
 
-function rangeContainsSkipTrivia(r1: TextRange, node: Node, file: SourceFile): boolean {
+function rangeContainsSkipTrivia(r1: TextRange, node: TextRange, file: SourceFile): boolean {
     return rangeContainsStartEnd(r1, skipTrivia(file.text, node.pos), node.end);
 }
 
-function collectTypeParameters(checker: TypeChecker, selection: TypeNode, enclosingNode: Node, file: SourceFile): TypeParameterDeclaration[] | undefined {
+function collectTypeParameters(checker: TypeChecker, selection: TypeNode | TypeNode[], enclosingNode: Node, file: SourceFile): TypeParameterDeclaration[] | undefined {
     const result: TypeParameterDeclaration[] = [];
+    const selectionRange = isArray(selection) ? { pos: selection[0].pos, end: selection[selection.length - 1].end } : selection;
+    if (isArray(selection)) {
+        selection.forEach(t => {
+            if (!visitor(t)) return undefined;

```

Comment:
I'm not sure I understand what's going on here: if `visitor` returns true, that means we want to skip collection of type parameters and so we want `collectTypeParameters` to return `undefined`, right?
It seems to me we're not doing that anymore here when `selection` is an array, since inside this `if` we're always returning the `result` array.